client/core/core.go

```diff
@@ -1207,6 +1210,41 @@ func (c *Core) handleUnbookOrderMsg(dc *dexConnection, msg *msgjson.Message) err
 	return ob.Unbook(&note)
 }
 
+// handleEpochOrderMsg is called when an epoch_order notification is
+// received.
+func (c *Core) handleEpochOrderMsg(dc *dexConnection, msg *msgjson.Message) error {
+	var note msgjson.EpochOrderNote
+	err := json.Unmarshal(msg.Payload, &note)
+	if err != nil {
+		return fmt.Errorf("epoch order note unmarshal error: %v", err)
+	}
+
+	// Ensure the associated dex lists the market referenced by the epoch note.
+	match := false
+	for _, mkt := range dc.markets {
+		if mkt.Display() == note.MarketID {
+			match = true
```

Comment:
Could just return here and get rid of the match variable.  Regarding it being an error, let me see if I understand: If Core is only setup for market X, but messages for market Y come in, that means the client has somehow registered on the `orderbook` route (which include the epoch order notes) for market Y or the server is misbehaving and sending messages that weren't what the client registered for.  Does that sound right?

Reply:
I'd say both cases outlined apply

Reply:
I should be breaking after setting `match` however.

Reply:
Oh, right, the check below is for `!match`.

Reply:
updated the commit, would like to squash it if it's all good now.

Reply:
TBH, I prefer not squashing until the review is approved, if at all.  It helps reviewers follow changes, and github is squashes for me on merge.